### PR TITLE
[14.0x.] Jenkins changes

### PR DESCRIPTION
**Pull request by pruivo**
* ~~Use JDK 21~~
* Increased memory from 1G to 1.5G
* Removed old stage

Using JDK 21 requires some code changes for another time :+1:

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -22,7 +22,7 @@ pipeline {
                     // Show the agent name in the build list
                     echo env.NODE_NAME
                     env.MAVEN_HOME = tool('Maven')
-                    env.MAVEN_OPTS = "-Xmx1g -XX:+HeapDumpOnOutOfMemoryError"
+                    env.MAVEN_OPTS = "-Xmx1500m -XX:+HeapDumpOnOutOfMemoryError"
```
Comment (ryanemerson) [inline Jenkinsfile:25]: We currently use 1200 on `main`. Should we increase there as well?

Comment (pruivo) [reply Jenkinsfile:25]: Yes, I'll do that.
 I've added one more change: the missing `url` and `repositoryId` :crossed_fingers:

Comment (pruivo) [reply Jenkinsfile:25]: created: https://github.com/infinispan/infinispan/pull/11204

**Comment by karesti** [comment]
@pruivo I merged the upstream PR, maybe too fast? this looks good, let me know

**Comment by pruivo** [comment]
@karesti it is fine. This change does not affect PR runs.

**Pull request merged**